akvo/rsr/models/project.py

```diff
@@ -981,13 +981,9 @@ def budget_total(self):
         return Project.objects.budget_total().get(pk=self.pk).budget_total
 
     def has_multiple_budget_currencies(self):
-        budget_items = BudgetItem.objects.filter(project__id=self.pk)
-        num_currencies = len(set([self.currency] + [c.currency if c.currency else self.currency for c in budget_items]))
-
-        if num_currencies > 1:
-            return True
-        else:
-            return False
+        budget_items = self.budget_items.all()
```

Comment:
It's a trade off between making an extra query vs. using the `budget_items` which were already pre-fetched.  We probably need to measure which one is better for the average case. I'll play around a little, with this code, and get back.